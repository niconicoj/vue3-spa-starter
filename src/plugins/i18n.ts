import { createI18n } from "vue-i18n";

import dayjs from "dayjs";
import "dayjs/locale/en";
import "dayjs/locale/fr";
import isLeapYear from "dayjs/plugin/isLeapYear";
import localizedFormat from "dayjs/plugin/localizedFormat";

import en from "../locales/en.json";
import fr from "../locales/fr.json";

dayjs.extend(isLeapYear);
dayjs.extend(localizedFormat);
dayjs.locale("en");

export const i18n = createI18n({
  locale: "fr", // set locale
  fallbackLocale: "en", // set fallback locale
  messages: {
    en,
    fr,
  },
});
