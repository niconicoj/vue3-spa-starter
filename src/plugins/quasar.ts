import { Quasar } from "quasar";

import "@quasar/extras/material-icons/material-icons.css";
import "quasar/src/css/index.sass";

export const quasar = {
  plugin: Quasar,
  opts: {},
};
