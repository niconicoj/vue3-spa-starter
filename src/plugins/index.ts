/**
 * plugins/index.ts
 *
 * Automatically included in `./src/main.ts`
 */

// Plugins
import { i18n } from "./i18n";
import { quasar } from "./quasar";
import { loadFonts } from "./webfontloader";

// Types
import type { App } from "vue";

export function registerPlugins(app: App) {
  loadFonts();
  app.use(i18n);
  app.use(quasar.plugin, quasar.opts);
}
