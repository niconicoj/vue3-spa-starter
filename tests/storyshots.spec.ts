import { composeStories } from "@storybook/testing-vue3";
import type { ContextedStory } from "@storybook/testing-vue3/dist/types";
import type { Meta, StoryFn } from "@storybook/vue3";
import { render } from "@testing-library/vue";
import { Quasar } from "quasar";
import { describe, expect, test, vi } from "vitest";
import { h } from "vue";

import { i18n } from "../src/plugins/i18n";

type StoryFile = {
  default: Meta;
  [name: string]: StoryFn | Meta;
};

console.error = vi.fn();

describe("Storyshots", async () => {
  const modules = await Promise.all(
    Object.values(import.meta.glob<StoryFile>(["../**/*.stories.ts"])).map(
      (fn) => fn()
    )
  );
  describe.each(
    modules.map((module) => ({ name: module.default.title, module }))
  )("$name", ({ name, module }) => {
    test
      .skipIf(name?.includes("NoStoryshots"))
      .each(
        Object.entries<ContextedStory<unknown>>(composeStories(module)).map(
          ([name, story]) => ({ name, story })
        )
      )("$name", ({ story }) => {
      const mounted = render(story(), {
        global: {
          plugins: [i18n, Quasar],
          components: { "router-link": h("a", { type: "primary" }) },
        },
      });
      expect(mounted.html()).toMatchSnapshot();
    });
  });
});
