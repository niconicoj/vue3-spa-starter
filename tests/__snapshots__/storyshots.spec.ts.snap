// Vitest Snapshot v1

exports[`Storyshots > Common/button > Default 1`] = `"<button class=\\"q-btn q-btn-item non-selectable no-outline q-btn--standard q-btn--rectangle q-btn--actionable q-focusable q-hoverable\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"q-focus-helper\\"></span><span class=\\"q-btn__content text-center col items-center q-anchor--skip justify-center row\\">cliquez ici</span></button>"`;

exports[`Storyshots > Movie/card > Default 1`] = `
"<div class=\\"q-card\\">
  <div class=\\"q-card__section q-card__section--vert\\">
    <div class=\\"text-h6\\">The Shawshank Redemption</div>
    <div class=\\"text-subtitle2\\">1994</div>
  </div>
</div>"
`;

exports[`Storyshots > Movie/list > Default 1`] = `
"<div class=\\"q-card\\" data-v-4a39eec0=\\"\\">
  <div class=\\"q-card__section q-card__section--vert\\" data-v-4a39eec0=\\"\\">
    <div class=\\"text-h6\\" data-v-4a39eec0=\\"\\">Films</div>
  </div>
  <div class=\\"q-card__section q-card__section--vert grid-2-col\\" data-v-4a39eec0=\\"\\">
    <div class=\\"q-card\\" data-v-4a39eec0=\\"\\">
      <div class=\\"q-card__section q-card__section--vert\\">
        <div class=\\"text-h6\\">The Shawshank Redemption</div>
        <div class=\\"text-subtitle2\\">1994</div>
      </div>
    </div>
    <div class=\\"q-card\\" data-v-4a39eec0=\\"\\">
      <div class=\\"q-card__section q-card__section--vert\\">
        <div class=\\"text-h6\\">The Godfather</div>
        <div class=\\"text-subtitle2\\">1972</div>
      </div>
    </div>
  </div>
</div>"
`;
